tests/thread_test.c

```diff
@@ -47,6 +50,12 @@ static int s_test_thread_creation_join_fn(struct aws_allocator *allocator, void
         aws_thread_get_detach_state(&thread),
         "thread state should have returned JOIN_COMPLETED");
 
+#ifdef AWS_PTHREAD_HAS_GETNAME
+    ASSERT_CURSOR_VALUE_STRING_EQUALS(
```

Comment:
debatable: we could add a new test, explicitly for setname/getname

I'd tacked the setname stuff into this existing test because there was no way to actually test the results before, I just wanted the code path the get exercised, but now we CAN check the results

Reply:
its currently in a test that does not do much besides joining and checking thread id, so i decided to just keep it in the same test